Reader: Fix card borders to account for display scale

frosty opened the pull request:
Fixes #12153, by reducing the width of the border on Reader cards.

![reader-borders-1](https://user-images.githubusercontent.com/4780/61566581-97219700-aa74-11e9-9302-6afd6569e88b.png)

**To test:**

* Build and run, and view a stream in the Reader
* Ensure that the cards have single pixel borders on both 2x and 3x screen types (e.g. iPhone XS and iPhone XS Max).

@etoledom, would you mind reviewing?

cc @mattmiklic 

Update release notes:

- [x] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.
(I didn't update release notes and I think it's so minor it can just be included in "Improved color scheme consistency" that I've added in another branch)

mattmiklic commented:
This looks good!

@etoledom good catch! In addition to reducing the line weight, any card that sits directly adjacent to the top toolbar should only have a border on the bottom, not the top or sides.

frosty commented:
Thanks for catching that @etoledom! I updated the border in that area too, and I extracted all of these `1.0 / UIScreen.main.scale` usages to a `.hairlineBorderWidth` property on `CGFloat`.
